### PR TITLE
feat: add gateway admin keycloak

### PR DESCRIPTION
**Pull request by bigglesandginger**
# Description

Adds admin-keycloak service to `dev` to allow development of UI.
Additionally updates the kubectl workaround to access Staging Keycloak now that manager1 is down

Resolves #(issue)

### CODE REVIEW

File: kubernetes/base/gateway/gateway-admin-keycloak-service.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+metadata:
+  labels:
+    software.uncharted.auth/component: admin-keycloak
+    software.uncharted.auth/environment: dev
+    software.uncharted.auth/tier: gateway
+  name: gateway-admin-keycloak
+spec:
+  ports:
+    - name: 8080-admin-keycloak-tcp
+      port: 8080
```
Comment (YohannParis) [inline kubernetes/base/gateway/gateway-admin-keycloak-service.yaml:12]: isn't port `8080` used by `hmi-client`?

File: kubernetes/overlays/dev/local/gateway/httpd/htdocs/index.html [hunk 2]
```diff
@@ -40,7 +50,10 @@ <h2>Terarium Auth</h2>
         <li>Adam the administrator <code>adam@test.io:asdf1ASDF</code></li>
         <li>Ursula the user <code>ursula@test.io:asdf1ASDF</code></li>
     </ul>
-    <p>Go to <a href="#" onclick="apiCall('/api/user/me')">/api/user/me</a>.</p>
-    <p>Go to <a href="#" onclick="apiCall('/api/admin')">/api/admin</a>.</p>
+    <h3>Admin Keycloak</h3>
+    <p>Get Roles <a href="#" onclick="apiCall('/adminapi/roles', 'GET')">/adminapi/roles</a>.</p>
+    <p>Go Users <a href="#" onclick="apiCall('/adminapi/users', 'GET')">/adminapi/users</a>.</p>
+    <p>Add role "user" to <input type="text" name="userId" id="userId" value="" /> <a href="#" onclick="apiCall('/adminapi/user/' + document.getElementById('userId').value + '/add-role?name=user', 'POST')">/adminapi/user/&lt;id&gt;/add-role?name=user</id></a>.</p>
+    <textarea id="responseArea"></textarea>
```
Comment (YohannParis) [inline kubernetes/overlays/dev/local/gateway/httpd/htdocs/index.html:57]: Is this how we give a keycloak user the role of "user"?
Does this works only for local, or is there a way we can config this page to available on _staging_ and _production_?

File: kubernetes/overlays/dev/local/gateway/gateway-httpd-deployment.yaml [hunk 3]
```diff
@@ -35,6 +35,8 @@ spec:
               value: http://host.docker.internal:3000/api/
             - name: LOCATION_APP_URL
               value: http://host.docker.internal:8080/app/
+            - name: ADMIN_KEYCLOAK_SERVICE_URL
+              value: http://gateway-admin-keycloak:8080/adminapi/
```
Comment (YohannParis) [inline kubernetes/overlays/dev/local/gateway/gateway-httpd-deployment.yaml:39]: With a domain name like this why even have a port number, even the adminapi.
Why not a simple `http://gateway-admin-keycloa/api/`?

**Review by YohannParis — COMMENTED** [review]
Who are those New1 and New2 users for?

File: kubernetes/base/gateway/gateway-admin-keycloak-service.yaml (hunk 1 above)
Comment (bigglesandginger) [reply kubernetes/base/gateway/gateway-admin-keycloak-service.yaml:12]: 8080 to the pod - not to the host - all services can run on the same port on their own pod - the only conflict area is when they have `type: LoadBalancer` and this service does not need (not really work) exposed to the host machine

File: kubernetes/overlays/dev/local/gateway/httpd/htdocs/index.html (hunk 2 above)
Comment (bigglesandginger) [reply kubernetes/overlays/dev/local/gateway/httpd/htdocs/index.html:57]: the index.html is for testing when the UI has not been updated, as for what system it works against - well that is governed by how the admin-keycloak service is defined and has access to - so the staging version deployed in staging would work against staging

File: kubernetes/overlays/dev/local/gateway/gateway-httpd-deployment.yaml (hunk 3 above)
Comment (bigglesandginger) [reply kubernetes/overlays/dev/local/gateway/gateway-httpd-deployment.yaml:39]: by default Java runs services on 8080 as port 80 is normally reserved (need sudo permissions).  Now you are correct that one could run the service within the container on port 80 with some effort (at least modifying an application.properties) but as this all internal glue it doesn't really matter one way or another - no users of the system would be aware of this

**Comment by bigglesandginger** [comment]
> Who are those New1 and New2 users for?

For testing

**Pull request merged**